test-resources-kafka/src/main/java/io/micronaut/testresources/kafka/KafkaTestResourceProvider.java

```diff
@@ -56,7 +63,11 @@ protected String getDefaultImageName() {
 
     @Override
     protected KafkaContainer createContainer(DockerImageName imageName, Map<String, Object> requestedProperties, Map<String, Object> testResourcesConfig) {
-        return new KafkaContainer(imageName);
+        if (isKraftMode(testResourcesConfig)) {
```

Comment:
I attempted to do a separate class that leveraged the statics in this class since they were the same.

That approach did not work. I'm guessing because they were both of `AbstractTestContainersProvider<KafkaContainer>` and the `KafkaKraftTestResourceProvider` was always ignored?

Reply:
I think we should take a slightly different approach to fix a discrepancy. The base image is different whether you use kraft mode or not. However, we also have the generic property to override the base image, so the `imageName` that you get may be different from what `getDefaultImageName()` returns. Therefore, I would add a check if the image name has been changed, and if so, keep it (but still enable kraft mode on that changed image if needed).

Otherwise, you lose the ability to override the version of the kraft image.

Reply:
@melix how do I check if the image name has been changed? I see how it's done in AbstractTestContainersProvider, but I cannot use `TestContainerMetadataSupport` inside `KafkaTestResourceProvider` because

> 'io. micronaut. testresources. testcontainers. TestContainerMetadataSupport' is not public in 'io. micronaut. testresources. testcontainers'. Cannot be accessed from outside package

Reply:
Whoops 😅